### DISCUSSION TITLE
partialClip doesn't have IsFinal property. How to determine end of stream.

### DISCUSSION THREAD

**Discussion by AntShestak** [RageAgainstThePixel/com.rest.elevenlabs#112] [Q&A] [answered]
TextToSpeechAsync() has a callbackL partialClipCallback: async (partialClip)

As per eleven labs API partialClip should have isFinal property
https://elevenlabs.io/docs/api-reference/websockets#streaming-output-audio

How to determine end of stream?

**Comment by StephenHodgson** [comment, marked as answer]
> How to determine end of stream?

When the task returns

**Reply by AntShestak** [reply]
Thank you very much! That works
